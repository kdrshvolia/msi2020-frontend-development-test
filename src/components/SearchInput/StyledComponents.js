import styled, { css } from 'styled-components';

const CustomInput = styled.input`
  width: 680px;
  height: 42px;
  border: 2px solid #333333;
  box-sizing: border-box;
  border-radius: 10px;
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  color: #ababab;
  padding: 10px 15px;
`;

export { CustomInput };
