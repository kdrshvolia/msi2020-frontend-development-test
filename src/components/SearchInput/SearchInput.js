import React from 'react';
import PropTypes from 'prop-types';
import { CustomInput, InputWrapper } from './StyledComponents';

const SearchInput = ({ placeholder, onChange, inputText }) => {
  const handleChange = (e) => {
    onChange(e.target.value);
  };

  return (
    <InputWrapper>
      <CustomInput
        type="text"
        placeholder={placeholder}
        onChange={handleChange}
        value={inputText}
      />
    </InputWrapper>
  );
};

SearchInput.propTypes = {
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  inputText: PropTypes.string.isRequired,
};

export default SearchInput;
