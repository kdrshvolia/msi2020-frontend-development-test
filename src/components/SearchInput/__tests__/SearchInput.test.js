import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchInput from '../SearchInput';

const isChanged = jest.fn();

describe('SearchInput', () => {
  it('Input text is changed', () => {
    const wrapper = mount(<SearchInput onChange={isChanged} />);
    const input = wrapper.find('input');
    const event = {
      target: {
        value: 'some text',
      },
    };
    input.simulate('change', event);
    expect(isChanged).toBeCalledWith(event.target.value);
  });

  it('Renders correctly', () => {
    const rendered = renderer.create(
      <SearchInput inputText="abc" onChange={isChanged} placeholder="text" />,
    );
    expect(rendered).toMatchSnapshot();
  });
});
