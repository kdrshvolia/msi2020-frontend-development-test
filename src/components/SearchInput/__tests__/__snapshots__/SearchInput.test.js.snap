// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Renders correctly 1`] = `
<div
  className="sc-AxjAm gtOyAx"
>
  <input
    className="sc-AxirZ iQbumS"
    onChange={[Function]}
    placeholder="text"
    type="text"
    value="abc"
  />
</div>
`;
